components/net/http_loader.rs

```diff
@@ -1508,88 +1514,29 @@ async fn http_network_or_cache_fetch(
             wait_for_cached_response(done_chan, &mut response).await;
         }
 
-        // Substep 5
+        // Step 10.5 If response is null, then:
         if response.is_none() {
+            // Step 10.5.1 Set response to forwardResponse.
+            let forward_response = response.insert(forward_response);
+
             if http_request.cache_mode != CacheMode::NoStore {
-                // Subsubstep 2, doing it first to avoid a clone of forward_response.
+                // Step 10.5.2 Store httpRequest and forwardResponse in httpCache, as per the
+                //             "Storing Responses in Caches" chapter of HTTP Caching.
+                // TODO: The spec doesn't explicitly tell us to check for CacheMode::NoStore.
+                //       Is this correct?
```

Comment:
Not the fetch spec, but the http cache spec does: https://httpwg.org/specs/rfc9111.html#response.cacheability